src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs

```diff
@@ -229,7 +192,8 @@ private Task OnStateChangedAsync(INetworkPeer peer, NetworkPeerState oldState)
                 || (peer.State == NetworkPeerState.Offline))
             {
                 peer.StateChanged.Unregister(this.OnStateChangedAsync);
-                this.ConnectedNetworkPeers.Remove(peer, "Peer disconnected");
+                this.peersById.TryRemove(peer.Connection.Id, out INetworkPeer unused);
+                peer.Dispose("Peer disconnected");
```

Comment:
I assume peer was already disposed if we reached this point so why call it again?

Reply:
Right ok so inside the collection Dispose was called so all good.